docs/DeveloperGuide.md

```diff
@@ -1,12 +1,406 @@
+
+### 1.2. UML Sequence Diagram
+
+![Sequence Diagram](SequenceDiagram.png)
+
+## 2. Implementation
+### 2.1. List
+The list command is mainly handled by the `ListCommand` class, which extends the `Command` class.
+
+![ListCommand.png](UML%2FList%2FListCommand.png)
+
+
+**Step 1**. When the user executes the command `list`, the `ParserHandler` will create a new `ListParser` object and pass to it the `Inventory` where the items to be listed are stored.
+
+![ListStep1.png](UML%2FList%2FListStep1.png)
+
+**Step 2**. The `run` method in `ListParser` overrides the `run` method in `Parser` to create a new `ListCommand` object, passing to it the relevant `Inventory`.
+
+![ListStep2.png](UML%2FList%2FListStep2.png)
+
+**Step 3**. The `run` method in `ListCommand` overrides the `run` method in `Command` and calls the `listItems` method. The `listItems` method checks if the inventory is empty. If the inventory is empty, the method prints a message to inform the user that there are no items in the inventory. Otherwise, the `printTable` method from the `Ui` object is called.
+
+![ListStep3.png](UML%2FList%2FListStep3.png)
+
+**Step 4.**. If the `printTable` method is called, it takes in an `ArrayList<Item> items` aas a parameter and prints out a table showing the name, UPC, quantity and price of all items in the inventory.
+
+
+
+
+
+
+
+### 2.2. Add
+The add command is mainly handled by the `AddCommand` and `AddParser` classes, which extends the `Command` and `Parser` 
+class respectively.
+#### 2.2.1. AddParser Class
+The `AddParser` class is a parser class that extends the `Parser` abstract class. It handles the "add" command by parsing 
+the user's input into separate input parameters using regular expressions. It then creates a new `Item` object with the 
+parsed parameters and creates a new `AddCommand` object with the `Inventory` object and `Item` object as parameters.
+
+The `AddParser` class has a constructor that takes in a `String` object representing the raw input from the user and an 
+`Inventory` object representing the inventory. It passes these parameters to the constructor of the `Parser` abstract class.
+
+The class contains an overridden `run()` method that first checks if the raw input is null or not. If it is null, it 
+throws a `MissingParametersException`. Otherwise, it uses a regular expression matcher to match the raw input to a 
+specific pattern. If the input doesn't match the pattern, it prints an error message and returns. If the input matches 
+the pattern, it creates a new `Item` object using the parsed parameters and creates a new `AddCommand` object using the 
+`Inventory` object and `Item` object. It then calls the `run()` method of the `AddCommand` object to execute the add command.
+![AddParser.png](UML%2FAdd%2FAddParser.png)
```

Comment:
I think that there is an additional return arrow between `AddParser` and `AddCommand`?